### ISSUE TITLE
Pre-trained models?

### ISSUE THREAD

**Issue by zuru** [palver7/CFLPytorch#1]
Hello, are there any pre-trained models available for the PyTorch implementation? 

If not, has this training implementation been verified to (closely?) reproduce the paper's results with the default hyperparameters (i.e. train for 100 epochs with a batch size of 4)?

A known issue is the incompatibility between PyTorch's and TensorFlow's weight decay which I suspect will influence results compatibility.

**Comment by palver7** [comment]
@zuru No, there are no pre-trained models for this implementation. I cannot reproduce the paper's performance and am still figuring out what bug is responsible for the performance gap. Thank you for pointing out about the incompatibility of the weight decay.

**Comment by zuru** [comment]
Setting weight decay to `0.0` improves training and the qualitative results are better indeed but the reported metrics (tried with snapshots @100 and @200 epochs):
```
EDGES: IoU: 0.153; Accuracy: 0.818; Precision: 0.230; Recall: 0.302; f1 score: 0.260
CORNERS: IoU: 0.046; Accuracy: 0.963; Precision: 0.099; Recall: 0.074; f1 score: nan
```  
indicate something is not there yet. 

The resulting corner maps though are noisy and corner detection does not behave very well.

Please let me know if you manage to identify the issue and reproduce the results.

**Comment by palver7** [comment]
@zuru Hi zuru, I have converted the pretrained weights from the original CFL to PyTorch and also implemented the model that can reproduce the results in the paper during inference. (there are a few differences in the metrics, they are close) The models have TFCFL in their names. I'll also put the link to the pretrained weights in README. Now, I still have not figured the exact training strategy used by Clara et.al so the training performance still does not come close to the inference  performance.
